0704 update v4 to v5 and new features (#1985)
emqtt/emq-docs-cn: en_US/getting-started/new-features.md

To get started, see [Create an EMQX Cluster](../deploy/cluster/create-cluster.md).

> ## Rolling Upgrades without downtime

Reviewer: ```suggestion
## Rolling Upgrades without Downtime
```

EMQX 5.1 introduces a set of command-line tools for backup and restore, allowing you to export data and configuration files from the built-in database as a compressed package and restore them to a new cluster.

> ::: details Usage example

Reviewer: ```suggestion
::: Detailed Usage Example
```

- Visualized data flows: Using [Flows](../dashboard/flows.md) users can clearly see how data flows from devices or clients through the rule engine
- Configuration updates during runtime: Hot update that takes effect immediately

> ## Overload protection, Limiter and Buffer queue for Bridges

Reviewer: ```suggestion
## Overload Protection, Limiter and Buffer Queue for Bridges
```

> New **Limiter** feature delivers a more accurate and layered rate control and ensures that the system operates under the expected workload, because it supports limiting client behavior at the client, listener, or node levels. The combination of Overload protection and Limiter prevents the clients from becoming too busy or receiving excessive request traffic and ensures stable system operation.

Reviewer: ```suggestion
The new **Limiter** feature enhances connection and messaging rate control by providing more precise and layered rate control options. It ensures that the system operates under the expected workloads by limiting the client behavior at the client, listener, or node levels. The combination of the Overload protection and Limiter features prevents the clients from becoming overwhelmed or receiving excessive request traffic and ensures stable system operation.
```

> We also added a generic buffer queue for all bridges which allows to buffer messages generated under stressful conditions. It can be configured to store messages in memory or disk cache when external resources are unavailable, for example, due to network fluctuations or service downtime. Messages will be sent after the service is restored. However, requests may expire in the buffer – this is a big difference comparing with v4. If the buffered data exceeds the limit, it will be discarded following the First In First Out (FIFO) rule.

Reviewer: ```suggestion
A generic buffer queue is also added for all bridges, allowing buffer messages generated under stressful conditions. This buffer can be configured to store messages in either memory or disk cache when external resources are unavailable, such as during network fluctuations or service downtime. Buffered messages will be sent once the service is restored. However, requests in the buffer may expire, which is a big difference compared to version 4. If the amount of buffered data exceeds the limit, it will be discarded following the First In First Out (FIFO) rule.
```

**Simplified configuration**

The `emqx.conf` configuration file has been changed to a concise and readable [HOCON](https://github.com/emqx/hocon) format, and contains commonly used configuration items by default, to improve the readability and maintainability.

**Improved REST API**

REST API is now compliant with the OpenAPI 3.0 specification, and comes with clear and rich API documentation.

**Rapid troubleshooting**

> Added more diagnostic tools such as slow subscriptions and online tracing so users can quickly troubleshoot issues in production.

Reviewer: ```suggestion
More diagnostic tools such as slow subscriptions and online tracing are added so users can quickly troubleshoot issues in production.
```

**Structured logs**

> Added support for machine (indexer) friendly structured logs in JSON format. Error logs are consistently tagged with 'msg' tokens to facilitate locating the cause of the problem.

Reviewer: ```suggestion
Machine (indexer) friendly structured logs in JSON format are supported. Error logs are consistently tagged with 'msg' tokens to facilitate locating the cause of the problem.
```

> **Flexible expansion and customization**

Reviewer: ```suggestion
**Flexible Expansion and Customization**
```

> Developed a new plugin architecture, with which users can compile, distribute, and install their extension plugins in the form of independent plugin packages to customize and extend the usage of EMQX.

Reviewer: ```suggestion
A new plugin architecture is developed, with which users can compile, distribute, and install their extension plugins in the form of independent plugin packages to customize and extend the usage of EMQX.
```
